### PR TITLE
chore(op-node): Add tests for Ecotone deposit transactions

### PR DESCRIPTION
**Pull request by danyalprout**
**Description**

Add uint tests for deposit transactions that are being submitted at the ecotone hardfork.

### CODE REVIEW

File: op-chain-ops/genesis/layer_two.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ func BuildL2Genesis(config *DeployConfig, l1StartBlock *types.Block) (*core.Gene
 	}
 
 	db, err = setupEcotone(config, db)
+	if err != nil {
+		return nil, err
+	}
```
Comment (EvanJRichard) [inline op-chain-ops/genesis/layer_two.go:81]: Wow, good save, thank you :skull:

File: op-node/rollup/derive/ecotone_upgrade_transactions.go [hunk 2]
```diff
@@ -108,12 +112,12 @@ func EcotoneNetworkUpgradeTransactions() ([]hexutil.Bytes, error) {
...
 
 	deployEIP4788, err := types.NewTx(&types.DepositTx{
-		From: common.HexToAddress("0x0B799C86a49DEeb90402691F1041aa3AF2d3C875"),
+		From: eip4788From,
 		// to is null
 		Mint:                big.NewInt(0),
 		Value:               big.NewInt(0),
 		Gas:                 0x3d090,
-		Data:                common.Hex2Bytes("0x60618060095f395ff33373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500"),
+		Data:                eip4788CreationData,
```
Comment (EvanJRichard) [inline op-node/rollup/derive/ecotone_upgrade_transactions.go:120]: Good cleanup :pray:

File: op-node/rollup/derive/ecotone_upgrade_transactions_test.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func toDepositTxn(t *testing.T, data hexutil.Bytes) (common.Address, *types.Transaction) {
+	txn := new(types.Transaction)
+	err := txn.UnmarshalBinary(data)
+	require.NoError(t, err)
+	require.Truef(t, txn.IsDepositTx(), "expected deposit txn, got %v", txn.Type())
+	assert.False(t, txn.IsSystemTx())
+
+	signer := types.NewLondonSigner(big.NewInt(420))
```
Comment (EvanJRichard) [inline op-node/rollup/derive/ecotone_upgrade_transactions_test.go:55]: is it worth making `big.NewInt(420)` a constant, or no? I could kind of go either way.

**Review by EvanJRichard — APPROVED** [review]
LGTM 😎 🚢

File: op-chain-ops/genesis/layer_two.go [hunk 4]
```diff
@@ -76,6 +76,9 @@ func BuildL2Genesis(config *DeployConfig, l1StartBlock *types.Block) (*core.Gene
 	}
 
 	db, err = setupEcotone(config, db)
```
Comment (tynes) [inline op-chain-ops/genesis/layer_two.go:78]: Hiding the branching logic inside of `setupEcotone` (which sounds like it always applies ecotone) is a bit unclear

File: op-node/rollup/derive/ecotone_upgrade_transactions.go [hunk 5]
```diff
@@ -23,7 +23,11 @@ var (
...
 	updateL1BlockProxySource   = UpgradeDepositSource{Intent: "Ecotone: L1 Block Proxy Update"}
 	updateGasPriceOracleSource = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Proxy Update"}
+	enableEcotoneSource        = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Set Ecotone"}
 	beaconRootsSource          = UpgradeDepositSource{Intent: "Ecotone: beacon block roots contract deployment"}
+
+	eip4788From         = common.HexToAddress("0x0B799C86a49DEeb90402691F1041aa3AF2d3C875")
+	eip4788CreationData = common.Hex2Bytes("0x60618060095f395ff33373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500")
 )
 
 func EcotoneNetworkUpgradeTransactions() ([]hexutil.Bytes, error) {
```
Comment (tynes) [inline op-node/rollup/derive/ecotone_upgrade_transactions.go:33]: We should be sure to use this function inside of the L2 genesis generation

File: op-chain-ops/genesis/layer_two.go (hunk 4 above)
Comment (EvanJRichard) [reply op-chain-ops/genesis/layer_two.go:78]: Will change in base branch

File: op-node/rollup/derive/ecotone_upgrade_transactions.go (hunk 5 above)
Comment (EvanJRichard) [reply op-node/rollup/derive/ecotone_upgrade_transactions.go:33]: Pulling this feedback into base PR

**Pull request merged**